sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/exceptions.py

```diff
@@ -0,0 +1,54 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import functools
+from typing import TYPE_CHECKING
+
+import six
+
+from azure.core.exceptions import ResourceExistsError, ResourceNotFoundError
+
+from ._generated.v2016_10_01.models import KeyVaultErrorException as KeyVaultError2016
+from ._generated.v7_0.models import KeyVaultErrorException as KeyVaultErrorV7
+
+if TYPE_CHECKING:
+    # pylint:disable=unused-import
+    from typing import Union
+
+    KeyVaultErrorExceptionType = Union[KeyVaultError2016, KeyVaultErrorV7]
+
+_error_map = {404: ResourceNotFoundError, 409: ResourceExistsError}
+
+
+def raise_from_key_vault_exception(exception):
```

Comment:
I'm curious as to why we couldn't use the `error_map` parameter instead of creating yet another way of mapping errors from autorest generated code?

Reply:
We could, I've reworked this to do that.